### DISCUSSION TITLE
`GroupBox` v2 - feature requests

### DISCUSSION THREAD

**Discussion by elParaguayo** [qtile/qtile#4365] [Ideas]
EDIT: Track progress here - https://github.com/elParaguayo/qtile-extras/pull/311

There have been numerous requests for changes to the `GroupBox` widget and I'm losing track of them. Post your requests down below and I'll see what I can do!

- [x] Line highlight for occupied workspaces (not just active one)
- [ ] Vertical bar support
- [x] Support list of colours (#4015)
- [x] Different highlight method for focused and unfocused groups (#2861)
- [x] Different text colours for active groups on different windows

**Comment by dranzer9944** [comment]
Please add line-highlight method and something like occupied_color for occupied workspaces.

**Comment by Crystal4276** [comment]
> Please add line-highlight method and something like occupied_color for occupied workspaces.

Yes ! I think i requested the same here :
https://github.com/qtile/qtile/issues/4009

**Comment by elParaguayo** [comment]
I'm going to try to set aside some time to look at this but it's hard!

There are so many possible scenarios that we need to cater for:
- Which screen we're on
- Whether that screen is focused
- Whether group is active or inactive
- Whether the group has windows or not

And then it's a question of can we set a different mode and colour for those different scenarios?

Struggling to see an easy way to make this configurable in a sensible way!

**Reply by elParaguayo** [reply]
Replying to myself...

I think approach would be to be have users set mode for the current group on a screen and set a mode for occupied groups that are not currently on any screen. I think we could allow multiple modes (e.g. block and line) fairly easily.

I'll have a go when I can.

**Reply by elParaguayo** [reply]
Update: making good progress on this. However, it will probably go into qtile-extras to start with. If it gets enough positive feedback we may look at deprecating the old one and replacing with this.

**Reply by elParaguayo** [reply]
Progress can be tracked here: https://github.com/elParaguayo/qtile-extras/pull/311

**Comment by elParaguayo** [comment]
@dranzer9944 @Crystal4276 Would you be interested in doing some beta testing of the new groupbox widget? It's in a PR in qtile-extras at the moment so you'd need to pull the relevant branch to install.

https://github.com/elParaguayo/qtile-extras/pull/311

I will post specific questions I'd like to be addressed in the PR.

Discussion for beta testing here: https://github.com/elParaguayo/qtile-extras/discussions/312

**Comment by TheHighGroundWins** [comment]
Add the option to hide unused tags, similar to the regular groupbox widget.

**Reply by elParaguayo** [reply]
Something like: `GroupBoxRule(visible=False).when(occupied=False, screen=GroupBoxRule.SCREEN_NONE)`

**Reply by TheHighGroundWins** [reply]
Works perfectly
